File: OCDemo/Classes/CustomTextViewController.m
//
//  CustomTextViewController.m
//  M80AttributedLabel
//
//  Created by amao on 5/21/14.
//  Copyright (c) 2014 www.xiangwangfeng.com. All rights reserved.
//

#import "CustomTextViewController.h"
#import "M80AttributedLabel.h"

@interface CustomTextViewController ()

@end

@implementation CustomTextViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    self.title = @"Custom Text";
    
    NSArray *fonts = @[[UIFont systemFontOfSize:12],[UIFont systemFontOfSize:13],[UIFont systemFontOfSize:17],[UIFont systemFontOfSize:25]];
    NSArray *colors= @[M80RGB(0x000000),M80RGB(0x0000FF),M80RGB(0x00FF00),M80RGB(0xFF0000)];
    
    
    M80AttributedLabel *label = [[M80AttributedLabel alloc]initWithFrame:CGRectZero];
    
    NSString *plainText = @"The release of iOS 7 brings a lot of new tools to the table for developers.";
    NSArray *components = [plainText componentsSeparatedByString:@" "];
    for (NSString *text in components)
    {
        NSMutableAttributedString *attributedText = [[NSMutableAttributedString alloc]initWithString:text];
        NSInteger index = arc4random() % 4;
        [attributedText m80_setFont:[fonts objectAtIndex:index]];
        [attributedText m80_setTextColor:[colors objectAtIndex:index]];
        
        [label appendAttributedText:attributedText];
        [label appendText:@" "];
    }
    label.frame     = CGRectInset(self.view.bounds,20,20);
    
    [self.view addSubview:label];

}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: SwiftDemo/Classes/CustomTextViewController.swift
//
//  CustomTextViewController.swift
//  SwiftDemo
//
//  Created by amao on 2016/10/12.
//  Copyright © 2016年 amao. All rights reserved.
//

import UIKit

class CustomTextViewController: BaseViewController {
    
    let fontSize : [CGFloat] = [12,13,17,25]
    let colors = [RGB(0x000000),RGB(0x0000ff),RGB(0x00ff00),RGB(0xff0000)]
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.title = "Custom Text"
        
        let label = M80AttributedLabel()
        let text = "The release of iOS 7 brings a lot of new tools to the table for developers."
        let components = text.components(separatedBy: " ")
        for component in components {
            let index = Int(arc4random() % 4)
            let attributedText = NSMutableAttributedString(string: component)
            attributedText.m80_setFont(UIFont.systemFont(ofSize: fontSize[index]))
            attributedText.m80_setTextColor(colors[index])
            
            label.appendAttributedText(attributedText)
            label.appendText(" ")
        }
        
        label.frame = self.view.bounds.insetBy(dx: 20, dy: 20)
        self.view.addSubview(label)

    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()

    }
    
}
